liberty-maven-plugin/src/main/java/io/openliberty/tools/maven/server/StartDebugMojoSupport.java

```diff
@@ -244,6 +246,13 @@ protected void copyConfigFiles() throws Exception {
         loadLibertyConfigFromProperties();
 
         if (configDirectory != null && configDirectory.exists()) {
+            // If mergeServerEnv is true, merge configDir server.env with generated server.env before the copy is made
+            File configDirServerEnv = new File(configDirectory, "server.env");
+            if(mergeServerEnv && configDirServerEnv.exists()){
+                File generatedServerEnv = new File(serverDirectory, "server.env");
+                mergeConfigDirServerEnv(configDirServerEnv, generatedServerEnv);
```

Comment:
Why are you doing this so differently than in ci.gradle? You should copy over everything from configDirectory except the configDirServerEnv, and then merge it with the default server.env just like you did in ci.gradle. Won't the copy below just overwrite what you just merged here?